include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -1349,11 +1340,19 @@ def TT_DescriptorGatherOp : TT_Op<"descriptor_gather", [MemoryEffects<[MemRead<G
   }];
 }
 
-def TT_DescriptorScatterOp : TT_Op<"descriptor_scatter", [
-    MemoryEffects<[MemRead<GlobalMemory>, MemWrite<GlobalMemory>]>,
-]> {
+def TT_DescriptorScatterOp : TT_Op<"descriptor_scatter"> {
+  let summary = "scatter multiple rows from a tensor into a descriptor";
+  let description = [{
+    The `tt.descriptor_scatter` op will be lowered to NVIDIA TMA
+    scatter operations on targets that support it.
+
+    `desc_ptr` is a pointer to the TMA descriptor allocated in global memory.
+    The descriptor block must have 1 row and the indices must be a 1D tensor.
+    Accordingly, the result is a 2D tensor multiple rows.
+  }];
+
   let arguments = (ins
-    TT_TensorDescType:$desc,
+    Arg<TT_TensorDescType, "", [MemWrite<GlobalMemory>]>:$desc,
```

Comment:
I wonder if this is correct to attach it to the argument as it isn't really a pointer?

Reply:
My thinking is that TensorDescType is a reference to global memory like MemDescType is a reference to shared or tensor memory. It is just a tiny bit weird because the tensor desciptor itself lives in global memory and contains a pointer to somewhere else in global memory, but we can consider the result of `make_tensor_descriptor` as aliasing the base pointer argument and it should be correct.

Reply:
If we wanted to distinguish between the tensor descriptor itself and the tensor data in gmem, we could introduce another resource called TensorMapMemory.

Reply:
TensorDescriptors are in theory immutable, so I think it's fine to not represent them as having memory effects. After lowering to `AsyncTMA` it's a different story though.